### PR TITLE
Fixed `istioctl x revision list` the output the data is not accurate of enabled components

### PR DESCRIPTION
**Pull request by nicole-lihui**
**Please provide a description of this PR:**
```bash
[root@controller-node-1 ~]# kubectl  get iop -n istio-system nicole-mesh -oyaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  labels:
    app.kubernetes.io/managed-by: Helm
  name: nicole-mesh
  namespace: istio-system
spec:
  #   ....
status:
  componentStatus:
    Base:
      status: HEALTHY
    IngressGateways:
      status: HEALTHY
    Pilot:
      status: HEALTHY
  status: HEALTHY

[root@controller-node-1 ~]# istioctl  x  revision list
REVISION TAG      ISTIO-OPERATOR-CR                     PROFILE REQD-COMPONENTS
default  <no-tag> istio-system/nicole-mesh default
# the data is not accurate of cloume `REQD-COMPONENTS`, need echo `base,ingressgateway,pilot`
```
DO:
- fix  the data is not accurate of cloume `REQD-COMPONENTS`
- optimize output format  of `-o table` code
- uniform output style with pkg `operator/pkg/name/name.go fuc: UserFacingComponentName()`,
  https://github.com/istio/istio/blob/e53f236e51a38ae06d64006ce66ae760ca794313/operator/pkg/name/name.go#L139-L148

Example:
```bash
[root@controller-node-1 ~]# istioctl  x  revision list
REVISION TAG      ISTIO-OPERATOR-CR                     PROFILE REQD-COMPONENTS
default  <no-tag> istio-system/nicole-mesh               default   Istio core
                                                                   Istiod
                                                                   Ingress gateways:istio-ingressgateway
```

### CODE REVIEW

File: releasenotes/notes/fix-istioctl-revision.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+releaseNotes:
+  - |
+    **Fixed** `istioctl x revision list` the data is not accurate  of column with `REQD-COMPONENTS`
```
Comment (ericvn) [inline releasenotes/notes/fix-istioctl-revision.yaml:6]:
```suggestion
    **Fixed** the `istioctl experimental revision list` `REQD-COMPONENTS` column data being incomplete and general output format.
```

**Review by ericvn — APPROVED** [review]
Release notes and tests look good.

**Review by hanxiaop — COMMENTED** [review]
Can you please show the root cause of this problem? I cannot reproduce in my environment.

**Comment by nicole-lihui** [comment]
> Can you please show the root cause of this problem? I cannot reproduce in my environment.

@hanxiaop 

The component list is detected based on the current `IstioOperator` CRD definition, but the actual deployed resources are merged from a custom IstioOperator and and. Istio `Profile` template. 

The custom IstioOperator only needs to configure the components to be customized, so there may be a component error in the middle.

example:
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: example-istiocontrolplane
spec:
  namespace: istio-system
  profile: default
  tag: 1.16.1
```
check： 
```bash
$istioctl x revision list
REVISION TAG      ISTIO-OPERATOR-CR                              PROFILE     REQD-COMPONENTS
default  <no-tag> istio-system/example-istiocontrolplane default.       
```
⚠️`REQD-COMPONENTS`  It definitely does not meet expectations.

File: istioctl/cmd/revision.go [hunk 2]
```diff
@@ -203,12 +205,16 @@ func revisionList(writer io.Writer, args *revisionArgs, logger clog.Logger) erro
 		if ri := revisions[rev]; ri == nil {
 			revisions[rev] = &tag.RevisionDescription{}
 		}
+		cs, err := getEnabledComponents(iop)
+		if err != nil {
+			return fmt.Errorf("error while get IstioOperator Components: %v", err)
```
Comment (hanxiaop) [inline istioctl/cmd/revision.go:210]:
```suggestion
			return fmt.Errorf("error while getting IstioOperator Components: %v", err)
```
nit: can you improve some of the error messsages to be more clearly?

**Review by hanxiaop — APPROVED** [review]

**Review by GregHanson — APPROVED** [review]

**Pull request merged**